### PR TITLE
Happy Thoughts API - Gabriella Bolin

### PR DESCRIPTION
**Pull request by Gabbi-89**
https://happythoughts-only.herokuapp.com/

Frontend site: https://happyminds.netlify.app/

### CODE REVIEW

File: server.js [hunk 1]
```diff
@@ -7,23 +7,118 @@ const mongoUrl = process.env.MONGO_URL || "mongodb://localhost/happyThoughts"
...
+  // Creates the initial database query
+  let databaseQuery = Thought.find();
+
+  // To search with sortField and sortOrder with queries: 
+  // (GET '/thoughts?sortField=${sortField}&sortOrder=${sortOrder}')
+  if (sortField) {
+    databaseQuery = databaseQuery.sort({ [sortField]: sortOrder === 'asc' ? 1 : -1 })
+  } else {
+    databaseQuery = databaseQuery.sort({ createdAt: 'desc' }).limit(20);
+  }
```
Comment (stjernberg) [inline server.js:76]: Nice to use sortField and sortOrder👌 This was new to me.

File: server.js [hunk 2]
```diff
@@ -7,23 +7,118 @@ const mongoUrl = process.env.MONGO_URL || "mongodb://localhost/happyThoughts"
...
+      errors: err.errors
+    });
+  }
+});
 
 // Start the server
 app.listen(port, () => {
   console.log(`Server running on http://localhost:${port}`)
-})
+});
```
Comment (stjernberg) [inline server.js:124]: Well written, clean code😃 I also like the comments that explains what all things do.

**Review by stjernberg — COMMENTED** [review]
Great job with this project Gabriella 😃 I think that you managed well to implement all the features according to the blue requirement. Also you put in extra effort to allow the user to add a name to the post and the ability to sort. Well done 👏

**Review by stjernberg — APPROVED** [review]

File: server.js (hunk 1 above)
Comment (KarolCislowski) [inline server.js:76]: When you created so nice functionality in your API why didn't you use it in your frontend?

**Review by KarolCislowski — APPROVED** [review]
Very nice project 👍🏻

File: server.js (hunk 1 above)
Comment (Gabbi-89) [reply server.js:76]: I didn't have time but I will really try to now! 😊

**Pull request closed without merging**